tests/tel2puml/find_unique_graphs/test_otel_data_holder.py

```diff
@@ -0,0 +1,184 @@
+"""Tests for OTel data holder classes."""
+
+import pytest
+
+from unittest.mock import patch
+from sqlalchemy import text
+
+from tel2puml.find_unique_graphs.otel_ingestion.otel_data_model import (
+    OTelEvent,
+    NodeModel,
+    SQLDataHolderConfig,
+)
+from tel2puml.find_unique_graphs.otel_ingestion.otel_data_holder import (
+    SQLDataHolder,
+)
+
+
+class TestSQLDataHolder:
+    """Tests for class SQLDataHolder."""
+
+    @staticmethod
+    def test_sql_data_holder_init(
+        mock_sql_config: SQLDataHolderConfig,
+    ) -> None:
+        """Tests the init method."""
+
+        holder = SQLDataHolder(mock_sql_config)
+        assert holder.batch_size == 10
+        assert holder.node_models_to_save == []
+        assert holder.node_relationships_to_save == []
+        with patch.object(holder, "create_db_tables") as mock_create_db_table:
+            holder.create_db_tables()
+            mock_create_db_table.assert_called_once()
+
+    @staticmethod
+    def test_create_db_tables(mock_sql_config: SQLDataHolderConfig) -> None:
+        """Tests the create_db_tables method."""
+
+        holder = SQLDataHolder(mock_sql_config)
+        with patch.object(
```

Comment:
I think here you should be actually checking the correct tables are created not just checking that the function has been called. Basically what are the expected tables and are they there